ppr-ui/src/utils/form-helper.ts

```diff
@@ -12,11 +12,31 @@ export function focusOnFirstError (formName: string): void {
 }
 
 // Scroll to first designated error on Information or Review page
-export async function scrollToFirstErrorComponent (): Promise<void> {
+export async function scrollToFirstErrorComponent (defaultIndex: number = 0): Promise<void> {
   setTimeout(() => {
     document.getElementsByClassName('border-error-left').length > 0 &&
       document
-        .getElementsByClassName('border-error-left')[0]
+        .getElementsByClassName('border-error-left')[defaultIndex]
         .scrollIntoView({ behavior: 'smooth' })
   }, 500)
 }
+
+/**
+ * Scroll to first visible instance of designated error class.
+ * Useful for multiple instances of error class across different views.
+ **/
+export async function scrollToFirstVisibleErrorComponent (): Promise<void> {
+  const errorElements = document.getElementsByClassName('border-error-left')
+  for (const [index, element] of Array.from(errorElements).entries()) {
+    if (isElementInViewport(element)) {
+      await scrollToFirstErrorComponent(index)
+      return
+    }
+  }
+}
```

Comment:
@dimak1 Really liked your simple scrollTo function, extended it a bit so we only scroll to visible elements. 
Context basically, in stepper flows where we have error instances across many steps in the dom, we only scroll to whats in our current viewport (ie step). 

Very simple, we just call this at root view layer and will scroll to any error border currently visible. Much improved way of scroll to vs try to track and feed identifiers to a function!

Reply:
Thank you! So the scroll to visible should be used where we have the stepper.

Reply:
Definitely! Seems to work good in that scenario.